src/core/snappingutils.cpp

```diff
@@ -32,11 +31,52 @@ void SnappingUtils::onMapSettingsUpdated()
   snap();
 }
 
+QgsPoint SnappingUtils::newPoint( const QgsPoint &snappedPoint, const QgsWkbTypes::Type wkbType )
```

Comment:
this code has been duplicated quite some time, not strictly required but would be cool to have this in QGIS core in QgsPoint as a static method there.

Reply:
The similar behavior is done around here https://github.com/qgis/QGIS/blob/master/src/gui/qgsmaptoolcapture.cpp#L760 but the evaluation of the WBK of the new point and the setting of the Z value are separated. Second function get's the point via mouse event. That's why it would request some rearrangements.
I'm open, but I suggest to keep it here...